### ISSUE TITLE
request for additional information in JSON statistics array

### ISSUE THREAD

**Issue by AnotherWorkingNerd** [inspec/inspec#1872]
### Description

This is a request for additional data be added to the JSON output. When using the command 
 `inspec exec test.rb -t ssh://someuser@example.com` you get the following lines at the end of the stdout:
```
Profile Summary: 10 successful, 1 failures, 1 skipped
Test Summary: 16 successful, 1 failures, 1 skipped 
```
However if you use the command like `inspec exec test.rb --format=json -t ssh://someuser@example.com` the json output contains a statistics array that contains a single element _duration_ and no summary information.

I am requesting that the 2 Summary lines be added to the JSON output. A timestamp would be nice too.

Thanks,
gmoore1 (on slack)

### InSpec and Platform Version

I am using 1.23.0 on OS X (macOS) but I don't believe it really matters in this case.

### Replication Case
With summary strings:
`inspec exec test.rb -t ssh://someuser@example.com` 

Without summary strings:
 `inspec exec test.rb --format=json -t ssh://someuser@example.com`

### Possible Solutions
I would suggest something simple like: 
```
 "statistics": {
    "duration": 4.459037
    "profile_summary": "10 successful, 1 failures, 1 skipped"
    "test_summary": "16 successful, 1 failures, 1 skipped"
    "timestamp": 1496187231 
  }
```
timestamp being the unix Epoch time `date +%s`.  Yea, we could get carried away here but I'd like to K.I.S.S.

**Comment by chris-rock** [comment]
@AnotherWorkingNerd Thank you for bring this question up. Could you tell us a little bit more about your use case? Why do you need those stats added, since they can be easily computed from the result set?

**Comment by adamleff** [comment]
I'm 👍  on this... anything to make it easier for users to consume the JSON payload is great. Even though it can be calculated, having us calculate it and return it in a "summary" section or similar makes it that much easier. Many of my favorite APIs do this rather than shifting the onus onto the receiver.

Down the line, when we make the CLI formatter based off of the JSON formatter, we'd need to calculate these stats anyways, whether we do it in the CLI formatter still or shift it to the JSON formatter. Seems like an easy win.

**Comment by AnotherWorkingNerd** [comment]
@chris-rock I can provide a longer explanation if needed. GCI is heavy into VMware and we have an automated vm creator thingy in vRO (I don't think I can go into a lot of detail about that) I am writing a vRO workflow that will get info about the new vm and using JSON pass this info to another machine that has InSpec on it, run a set of machine role specific tests/profiles, update the json object with the JSON results and send it back to vRO. The profile/test summary is an easy way to determine if the vm should be tagged good or bad. if the vm doesn't make it through testing then a human can look at the rest of the results and do an RCA and take whatever corrective action is needed. 
Basically, its a whole lot easier to check 1 or 2 elements than iterating through the controls array to calculate pass/fail criteria.

Yes I know this is a good use case for Automate and we have a license for it but its not installed and because we are using hosted chef server at the moment its much more then "just install Automate".

**Comment by chris-rock** [comment]
@AnotherWorkingNerd That is a great use case and it helps me to understand your needs better. Thank you for sharing. Do you think something like the following would also work?

```
"statistics": {
    "controls": {
      "total": 113,
      "passed": {
        "total": 22
      },
      "skipped": {
        "total": 68
      },
      "failed": {
        "total": 23,
        "minor": 0,
        "major": 0,
        "critical": 23
      }
    }
]
```

Do you need the stats for all profiles or is a report level fine? Just another idea that we could go: InSpec could show stats based on a saved report: `inspec stats reports.json`

**Comment by AnotherWorkingNerd** [comment]
@chris-rock Actually that was going to be my original suggestion!  That would defiantly work. :-)

**Comment by adamleff** [comment]
@chris-rock if we decide to implement #1825, we'd likely need to calculate this info as well at report-time.

**Comment by AnotherWorkingNerd** [comment]
as well as documenting what major and minor failures are. Which would help me A WHOLE LOT!. if there are major failures, fail the vm if only minor failures notify a human.

**Comment by clintoncwolfe** [comment]
Related: #3667

**Comment by EasyAsABC123** [comment]
@adamleff @chris-rock what is the word on this? seems like it would be done by now..

**Comment by kekaichinose** [comment]
This issue has aged and we do not have any plans to address this in the near future.

We apologize that we have not addressed your issue, but if it's still something you’re interested in solving, you’re welcome to submit a PR for it. If you decide to submit a PR, please reference this issue so we can re-open and track it and the PR to completion.

Thank you!
__Chef InSpec Team__
